packages/ui/cypress/support/page-objects/settingsPage.ts

```diff
@@ -0,0 +1,7 @@
+export const settingsPage = {
+  watchAccountAccountInput: () => cy.findByLabelText('Address'),
+  addButton: () => cy.findByRole('button', { name: /add/i }),
+  multisigIcon: () => cy.get('.identicon'),
+  multisigNameLabel: () => cy.get('.multisigName'),
+  multisigAddressLabel: () => cy.get('.multisigAddress')
```

Comment:
Hey @Lykhoyda, I would usually just add a data-cy tag to these multisig elements above

<img width="271" alt="CleanShot 2023-09-06 at 22 48 11@2x" src="https://github.com/ChainSafe/Multix/assets/3982665/421d4a60-7cd3-4d0e-bed2-be3e1eb75466">

I tried using findByRole but these elements don't produce anything to locate by. The only thing I could use without adding a tag is the class name....but that's a Cypress locator, not the testing playground stuff. 

It works fine but I was under the impression that locating elements via class names (with the default cypress locators) was inferior to other locator techniques so didn't want to leave it as is. Do you have any recommendations?

Reply:
@asnaith  It might also happen that there is no meaningful selector and as a last resort, we can use classes to query. However, we can be more specific here, for example, the parent component has a role "List" and we can search for classes inside this container with the "within" method (https://testing-library.com/docs/dom-testing-library/api-within/)

![Screenshot 2023-09-07 at 11 48 56](https://github.com/ChainSafe/Multix/assets/8141917/570f1a62-c920-462c-b2de-19d2defc1d6e)

So it will be something like that:
```
cy.findByRole('list').within(() => {
  cy.get('.identicon')
})
```

There are rules about the priority of selectors (https://testing-library.com/docs/queries/about#priority) and if you can't come up with a good selector you can use "data-testid" which is similar to cypress approach with id

```js
// In react app

            <NameStyled
              data-testid={'multisigName'}
              className="multisigName"
            >
              {localName || mainDisplay}
            </NameStyled>

```

```js
cy.findByTestId('multisigName')
```

Reply:
@Lykhoyda ah ok, thank you. 

So just to confirm - when we do have to use a `data-testid` we'd do that instead of `data-cy` so that we can use `getByTestId` from the testing library (rather than having a mix of locators from testing library and cypress)?